app/models/pipeline_run_stage.rb

```diff
@@ -53,8 +59,26 @@ def failed?
     job_status == STATUS_FAILED
   end
 
-  def succeeded? # The whole thing completed successfully
-    db_load_status == 1
+  def last_output_in_stage
+    basename = "#{job_id}.done"
+    s3_folder = case name
+                when HOST_FILTERING_STAGE_NAME
+                  pipeline_run.host_filter_output_s3_path
+                when ALIGNMENT_STAGE_NAME
+                  pipeline_run.alignment_output_s3_path
+                when POSTPROCESS_STAGE_NAME
+                  pipeline_run.postprocess_output_s3_path
+                end
+    "#{s3_folder}/#{basename}"
+  end
+
+  def succeeded?
+    # Return whether stage succeeded AND record the result in the database if it is not there already
+    answer = pipeline_run.file_generated_since_run(last_output_in_stage)
```

Comment:
Why does it have to be "generated since last run"?   We have versioning now, if it was generated in a previous run, that's fine.  In fact what would happen with your change if you drop this is that rerunning from top when lazy results exist for host filtering will cause the host filtering batch command to be killed immediately which is fantastic as then the alignment would start.

Reply:
The only thing is that in your pipeline status reporting that might show up as a "failed" status for the rerun, but that's no problem, since there would have been a "succeeded" status for the previous run (the one that produced the results).

Reply:
I see your point and it seems like a good change to make. What if lazy_run is set to False in the pipeline though? As is the case for the entire postprocessing stage? The web app change would supersede that. Maybe that's fine